python/ray/serve/_private/application_state.py

```diff
@@ -75,11 +123,18 @@ def update(self):
 
         if self.status == ApplicationStatus.DELETING:
             mark_delete = True
+            # Application won't be deleted until all deployments get cleaned up
             for name in self.get_all_deployments():
                 if self.deployment_state_manager.get_deployment(name):
+                    logger.warn(
+                        f"Deleting deployment {name} from application {self.name}."
+                    )
                     mark_delete = False
                     break
+            if self.deployments_to_delete:
+                mark_delete = False
             self.to_be_deleted = mark_delete
```

Comment:
Is it possible to rename `to_be_deleted`, it's a bit confusing to name it from the perspective of the `ApplicationStateManager` when still inside `ApplicationState` - maybe `done_deleting_deployments` or `all_deployments_deleted`?

Reply:
deployments_to_delete is a set() variable to keep track all deployments, it is not boolean variable.

Reply:
Oh, I meant the boolean `to_be_deleted`